docs/IcepeakProtocol.md

```diff
@@ -19,3 +29,42 @@ Example:
 3. The server immediately sends the current value of `/path/to/channel`, e.g. `{"status": "updating"}`
 4. The value gets updated to `{"status": "done"}` (by another process)
 5. The server then immediately sends the updated value to all clients listening to `/path/to/channel`
+
+### Modifying data
+
+Data on Icepeak can be modified by either a PUT request or a DELETE request. The PUT will introduce
+or overwrite new information to the Icepeak server and DELETE will remove information from the server.
+In case some information changes, all information that is changed will be send to every 
+client that is listening to that channel. 
+
+- `Put` writes a JSON value to a given path in the JSON structure.
+
+  - If the path descends into a JSON value that is not an object, that value is
+    replaced by an object that just contains the newly created path.
+  - If a value along the path is already an object, the field specified by the
+    path is added to it.
+
+- `Delete` removes a JSON value at a given path.
+
+  - If the path does not exist, nothing happens.
+  - If the path refers to a field in an object, that field is removed from the
+    object.
+
+Both of these operations have an optional parameter, called `durable`, which if set,
+ensures that the action is completed before the response code 202 (accepted) is
+send back to the client. This ensures that no stale data is possible to get via
+a get call. 
+
+Example if `durable` is **not** set:
+1. The client sends a PUT command to the server with a new value for `foo`. 
+2. The server responds with a 202 (accepted) response immidiatly, before completing the action.
+3. A GET request is send to the server, requesting the value for `foo`, but `foo` is not yet updated on the server.
```

Comment:
This can always happen if the GET request is sent from another server, as a GET request will not go through the Core event loop.
For the purpose of demonstrating the difference, it will be more interesting to consider a GET request that has been sent from the same client that also sent the PUT command.

Reply:
And actually, even the latter is not guaranteed if multiple concurrent users send a PUT request to the same path on Icepeak with different values.

So a durable request doesn't really guarantee that you `GET` the same value that you `PUT` before, but rather that if the `PUT` response has been received, the client can be sure it's update has been processed.
Conversely, it could very well be that the request was processed, but that the client didn't receive the response (e.g. due to a network issue).